packages/ui/components/ui/json-viewer.tsx

```diff
@@ -1,128 +1,40 @@
-
-      // nested hidden objects stay open
-      return true;
-    },
-    [expandAll],
-  );
-
   return (
     <div className='mt-5 rounded bg-black p-5'>
-      <div className='relative w-full'>
-        <div className='absolute right-0 top-0 w-full'>
-          <div className='relative flex items-end justify-end gap-2 text-muted-foreground'>
-            <ExpandAllIconButton expandAll={expandAll} setExpandAll={setExpandAll} />
-            <CopyToClipboardIconButton text={JSON.stringify(jsonObj)} />
-          </div>
-        </div>
-      </div>
-      <JsonView
-        data={jsonObj}
-        shouldExpandNode={shouldExpandNode}
-        style={{
-          container: cn(
-            'font-mono',
-            'mr-[3em]', // pad on the right side, for balance
-            'text-ellipsis', // visually truncate long strings on the right side
-            '[&>div>span:first-child]:hidden', // hide first collapse arrow
-          ),
-          basicChildStyle: cn(
-            'text-gray-500',
-            'break-keep truncate', // don't break strings, visually truncate
-            'pl-[1em]', // indent each level
-            // compact display if no children
-            '[&:not(:has(div))]:block',
-            'only-of-type:[&:not(:has(div))]:p-0',
-            'only-of-type:[&:not(:has(div>div))]:inline',
-            'only-of-type:[&:not(:has(div>div))]:*:inline',
-            '[&>span:last-child]:mr-2', // space after last item, for compact display
-          ),
-          label: cn(
-            'text-gray-200',
-            'inline',
-            'mr-2', // space after label
-          ),
-          nullValue: 'text-red-600',
-          undefinedValue: 'text-red-600',
-          noQuotesForStringValues: true, // quotes will be styled
-          stringValue: cn(
-            'text-amber-600',
-            'select-all', // entire string selected on click
-            // quotes from style, so they don't get selected
-            "before:content-['“'] before:text-gray-500",
-            "after:content-['”'] after:text-gray-500",
-          ),
-          booleanValue: 'text-purple-600',
-          numberValue: 'text-teal-200',
-          otherValue: 'text-blue-600',
-          punctuation: cn(
-            'text-gray-500',
-            'inline text-sm',
-            'has-[~div]:mr-2', // space if next item is a label
-          ),
-          collapseIcon: cn(
-            'text-teal-600 text-lg',
-            "has-[~div>div:nth-of-type(2)]:after:content-['▼']",
-            'leading-[0] w-0 inline-block relative -left-3',
-          ),
-          expandIcon: 'hidden', // use collapsedContent ellipsis to expand
-          collapsedContent: "text-teal-600 text-xs leading-[0] after:content-['•••'] after:mx-2",
-        }}
+      <TextaJsonViewer
+        value={jsonObj}
+        style={{ fontFamily: 'Iosevka Term' }}
+        theme={customTheme}
+        rootName={false}
+        enableClipboard={true}
+        defaultInspectDepth={2}
```

Comment:
if it's possible to default to expand things below or at a certain distance below the collapsed level, i would prefer that

Reply:
If I understand you correctly, this lib does not provide a way to customize how deep an expand is